src/pip/_internal/network/download.py

```diff
@@ -34,11 +33,7 @@ def _prepare_download(
 ) -> Iterable[bytes]:
     total_length = _get_http_response_size(resp)
 
-    if link.netloc == PyPI.file_storage_domain:
-        url = link.show_url
-    else:
-        url = link.url_without_fragment
-
+    url = link.show_url
```

Comment:
This is the only part of the change that I'm somewhat fine with, but I wonder if show_url should also mention the domain. That's something we can address in a follow up though.